### DISCUSSION TITLE
v0.17.0

### DISCUSSION THREAD

**Discussion by Julien-cpsn** [Julien-cpsn/ATAC#103] [Release feedback]
# A big wait for a big reward 🎉

![v0.17.0 showcase](https://github.com/Julien-cpsn/ATAC/blob/main/gifs/atac_v0.17.0.gif)

## Added

- ⌨ | ATAC is now **FULLY usable by CLI**!!!!! (please use it a lot, it cost me 40h+ of my life)
  - You can find the list of all commands below
  - The `-h|--help` flag or help command is available everywhere, abuse it if you need
  - A great start can be to use the `atac completions <SHELL_TYPE>` command to generate completion files
  - Another great start could be to generate the man pages with the `atac man` command
  - Ya like cURL? Behold `atac try` which is a one-shot request sender (like cURL)
- 📜 | Totally **reworked the log system**, now almost everything is getting logged. Also, the log file now really contains logs (incredible I know). @v3rmine @i5.650
  - Modulate the verbosity with `-v|vv|vvv|vvvv` or `-q`
  - Deactivate ANSI formatting with `--no-ansi-log`
- 🌱 | Added **reserved environment variables**, they will automatically be replaced by their value 
  - `{{NOW}}`
  - `{{TIMESTAMP}}`
  - `{{UUIDv4}}`
  - `{{UUIDv7}}`
- ☠ | Possibility to **allow invalid certificates** (https://github.com/Julien-cpsn/ATAC/pull/99). @ficolas2
Please update your collections as follows:
```json
{
...request...
      "settings": {
        "use_config_proxy": true,
        "allow_redirects": true,
        "store_received_cookies": true,
        "pretty_print_response_content": true,
        "accept_invalid_certs": false
      }
}
```

## Improved

- 🔥 | Added multithreaded parallelization to a lot of iterators (for loops). Now parses, finds and collects data faster! (**blazingly fast**)
- 🤷 | Improved auth data structure
  - Basic auth
```json
{
...request...
    "auth": {
        "basic_auth": {
            "username": "user",
            "password": "pass"
        }
    }
}
```
  - Bearer token
```json
{
...request...
    "auth": {
        "bearer_token": {
            "token": "my_bearer="
        }
    }
}
```
- 🖊 | Improved the [documentation](https://github.com/Julien-cpsn/ATAC-website). If something is missing, please contribute! @Zacktamondo @Cephra
- ⬆ | Updated all dependencies

## Fixed

- Body environment variables not getting replaced by their value (https://github.com/Julien-cpsn/ATAC/pull/101) @anussel5559
- Enormous cosmetic change, thanks @orhun (https://github.com/Julien-cpsn/ATAC/pull/89)

- - -

List of all the commands:

```
atac
  - collection
      - list
      - info
      - new
      - delete
      - rename
      - send (all requests from the collection)
  - request
      - info
      - new
      - delete
      - rename
      - url
      - method
      - params
      - auth
      - headers
      - body
      - scripts
      - send
      - settings
  - try
  - env
      - info
      - key
          - get
          - add
          - set
          - delete
          - rename
  - import
      - postman
      - curl
 - completions
      * bash, powershell, fish
 - man
```

<hr /><em>This discussion was created from the release <a href='https://github.com/Julien-cpsn/ATAC/releases/tag/v0.17.0'>v0.17.0</a>.</em>

**Comment by orhun** [comment]
I mentioned this release in my talk today :)

- https://media.ccc.de/v/froscon2024-3147-renaissance_of_terminal_user_interfaces_with_rust#t=2007

> Enormous cosmetic change, thanks @orhun

also lmao
